packages/core/src/api/blockManipulation/commands/splitBlock/splitBlock.ts

```diff
@@ -1,81 +1,50 @@
-import { Fragment, Slice } from "prosemirror-model";
-import { EditorState, TextSelection } from "prosemirror-state";
+import { EditorState } from "prosemirror-state";
 
 import { getBlockInfoFromPos } from "../../../getBlockInfoFromPos.js";
 
-export const splitBlockCommand =
-  (posInBlock: number, keepType?: boolean, keepProps?: boolean) =>
-  ({
+export const splitBlockCommand = (
```

Comment:
Couple of things to discuss

1) there seem to be some bugs with this implementation (e.g.: enter end of line) (suppose you might already be aware of this)
2) wondering why in the new version the entire blockgroup is deleted + recreated, but in the old version not
3) wondering if we can't used a (modified) https://github.com/ProseMirror/prosemirror-commands/blob/43c118ffc9d1e8a097a320d07e430e2faf861778/src/commands.ts#L357 but that uses tr.split with depth 2 to also split blockcontainer (see https://github.com/ProseMirror/prosemirror-transform/blob/814bd305b17363a975ffc5ccc55586c875e6d9c6/src/structure.ts#L211)